application/messages/messages.go

```diff
@@ -17,3 +31,62 @@ const (
 	MessageTemplateItemRevisionMember   = "item_revision_member"
 	MessageTemplateItemDeniedMember     = "item_denied_member"
 )
+
+type MessageData render.Data
+
+func newEmailMessageData() MessageData {
+	m := MessageData{}
+
+	m["uiURL"] = domain.Env.UIURL
+	m["appName"] = domain.Env.AppName
+
+	return m
+}
+
+func (m MessageData) addItemData(item models.Item) {
+	if m == nil {
+		m = map[string]interface{}{}
+	}
+
+	m["itemURL"] = fmt.Sprintf("%s/items/%s", domain.Env.UIURL, item.ID)
+	m["itemName"] = item.Name
+	return
+}
+
+func (m MessageData) renderHTML(template string) string {
+	bodyBuf := &bytes.Buffer{}
+	data := render.Data(m)
+	if err := notifications.EmailRenderer.HTML(template).Render(bodyBuf, data); err != nil {
+		panic("error rendering message body - " + err.Error())
+	}
+	return bodyBuf.String()
+}
+
+func SendQueuedNotifications(tx *pop.Connection) {
+	var notnUsers models.NotificationUsers
+	if err := notnUsers.GetQueuedEmails(tx); err != nil {
+		panic(err.Error())
+	}
+
+	for _, n := range notnUsers {
+		n.Load(tx)
+		msg := notifications.NewEmailMessage()
+		msg.ToName = n.User.Name()
+		msg.ToEmail = n.EmailAddress
+		msg.Body = n.Notification.Body
+		msg.Subject = n.Notification.Subject
+
+		if err := notifications.Send(msg); err != nil {
+			domain.ErrLogger.Printf("error sending queued notification email, %s", err)
+			// TODO figure out how to have a longer delay each time
+			n.SendAfterUTC = time.Now().UTC().Add(time.Minute * 10)
```

Comment:
Maybe add a const or even an Env for the 10 minute value.

Reply:
:+1:

Reply:
Use an exponential backoff, something like:
```suggestion
			delay := n.SendAttemptCount^2 * time.Minute
			if n.SendAttemptCount > 10 {
				delay = 100 * time.Minute
			}
			n.SendAfterUTC = time.Now().UTC().Add(delay)
```